### DISCUSSION TITLE
Looking for a deeper explanation of how online prediction is meant to work

### DISCUSSION THREAD

**Discussion by rinnadom** [feast-dev/feast#1557] [Q&A]
I recently started investigating Feast, and looking through the documentation I think it could be a useful tool for my team. The only thing I'm having a difficult time understanding is how the "online" functionality is meant to work in practice.

I understand how the basic example in the documentation works, but I'm having trouble understanding what this means for real-time predictions. If a client has a new data point that they want to get a prediction on, it seems like they have to write that point to a data warehouse, run the feature transform steps, materialize it in the online store, and then make their prediction. This seems like a lot of steps, and I'd imagine has some overhead.

Is Feast meant to be used for real-time predictions, or am I misunderstanding the purpose of the "online" functionality?

**Comment by woop** [comment]
Hey @rinnadom, this comment somehow escaped us. 

Yes it's true that there are lots of steps in productionizing data for ML. Feast isn't meant to solve all of those problems, at least not now. Right now we are only focused on taking data that you have already processed and serving it from an online store. Often teams need to access their analytics data at low latency and in a consistent way to their offline data (for training). Feast helps with that aspect. 

So typically you'd run Feast side-by-side with a model serving system.

**Reply by rinnadom** [reply]
@woop Thanks for your reply, that helps clear things up for me!
